### ISSUE TITLE
builtin provider does not respect filepath filtering from chained conditions

### ISSUE THREAD

**Issue by tsanders-rh** [konveyor/analyzer-lsp#1005]
## Problem Description

The `builtin` provider does not respect filepath filtering from the `from:` clause in chained conditions, unlike the gRPC providers (nodejs, java, dotnet) which were fixed in PR #1004.

## Test Case

When using a chained rule like this:

```yaml
when:
  and:
    - as: tsFiles
      nodejs.referenced:
        pattern: Button
        ignore: true
    - builtin.fileContent:
        filepaths: '{{tsFiles.filepaths}}'
        pattern: isSelectable
        from: tsFiles
```

**Expected Behavior:**
The `builtin.fileContent` condition should only search in the files identified by the `tsFiles` variable from the first condition (respecting the `from: tsFiles` scope).

**Actual Behavior:**
The `builtin.fileContent` provider searches ALL files in the codebase, ignoring the filepath scope passed via `from: tsFiles`.

## Root Cause

The builtin provider does not call `ProviderContext.GetScopedFilepaths()` to extract and apply filepath filtering from chained conditions.

## Proposed Solution

Implement filepath filtering in the builtin provider similar to what was done for gRPC providers in PR #1004:

1. Extract scoped filepaths using `ProviderContext.GetScopedFilepaths()`
2. Build maps for O(1) lookups of included/excluded paths
3. Filter results to only include files within scope
4. Handle URI normalization for cross-platform compatibility

Reference implementation: See changes to `nodejs/service_client.go`, `java/service_client.go`, and `dotnet/service_client.go` in PR #1004.

## Related

- Closes #1002 (partially - gRPC providers fixed, builtin provider remains)
- Related to PR #1004 (fixed gRPC providers)

## Files to Modify

The builtin provider implementation that needs filepath filtering support (likely in the provider's evaluation logic where it processes `filepaths` parameter).
